cmd/pulumi-test-language/interface.go

```diff
@@ -136,6 +145,59 @@ func (eng *languageTestServer) Done() error {
 	return <-eng.done
 }
 
+func (eng *languageTestServer) Log(_ context.Context, req *pulumirpc.LogRequest) (*pbempty.Empty, error) {
+	eng.logLock.Lock()
+	defer eng.logLock.Unlock()
+
+	var sev diag.Severity
+	switch req.Severity {
+	case pulumirpc.LogSeverity_DEBUG:
+		sev = diag.Debug
+	case pulumirpc.LogSeverity_INFO:
+		sev = diag.Info
+	case pulumirpc.LogSeverity_WARNING:
+		sev = diag.Warning
+	case pulumirpc.LogSeverity_ERROR:
+		sev = diag.Error
+	default:
+		return nil, fmt.Errorf("Unrecognized logging severity: %v", req.Severity)
+	}
+
+	message := req.Message
+	if os.Getenv("PULUMI_LANGUAGE_TEST_SHOW_FULL_OUTPUT") != "true" {
+		// Cut down logs so they don't overwhelm the test output
+		if len(message) > 2048 {
+			message = message[:2048] + "... (truncated, run with PULUMI_LANGUAGE_TEST_SHOW_FULL_OUTPUT=true to see full logs))"
+		}
+	}
+
+	if eng.previousMessage == message {
+		eng.logRepeat++
+		return &pbempty.Empty{}, nil
+	}
+
+	if eng.logRepeat > 1 {
+		_, err := fmt.Fprintf(os.Stderr, "Last message repeated %d times\n", eng.logRepeat)
+		if err != nil {
+			return nil, err
+		}
+	}
+	eng.logRepeat = 1
+	eng.previousMessage = message
+
+	var err error
+	if req.StreamId != 0 {
+		_, err = fmt.Fprintf(os.Stderr, "(%d) %s[%s]: %s", req.StreamId, sev, req.Urn, message)
+	} else {
+		_, err = fmt.Fprintf(os.Stderr, "%s[%s]: %s", sev, req.Urn, message)
```

Comment:
Hrm these logs are the most painful part to dig through when running conformance tests (they are not per test, so we always just spew out all of them, and they are 99% of the time useless noise).  Just printing them to stderr probably makes that worse, especially in CI.  I don't have a smart suggestion here unfortunately, but this would be nice to improve instead of making worse.

Reply:
This is in the test host process, which all the tests capture stderr and forward to t.Log, so there shouldn't be any observable change from what we have today.